modules/c++/plugin/source/ErrorHandler.cpp

```diff
@@ -23,7 +23,7 @@
 #include "plugin/ErrorHandler.h"
 
 plugin::DefaultErrorHandler::DefaultErrorHandler(logging::Logger* logger) :
-	mLogger(logger)
+	mLogger(mem::SharedPtr<logging::Logger>(logger))
```

Comment:
This seems bad.  If we're going to use `SharedPtr` it needs to be passed in that way as well, otherwise this doesn't buy us anything and will likely blow up.